src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,49 @@
+@import '../utils/functions';
+@import '../utils/variables';
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap-size;
+  width: 750px;
+  padding: 10px;
+
+  $days: "mon" 0, "tue" 1, "wed" 2, "thu" 3, "fri" 4, "sat" 5, "sun" 6;
+
+  @each $day, $number-of-day in $days {
+    &--start-day--#{$day} .calendar__day:first-child {
+      margin-left: get-margin-size($number-of-day);
+    }
+  }
+
+  @for $length from 28 through 31 {
+    &--month-length--#{$length} .calendar__day:nth-child(n + #{$length + 1}) {
+      visibility: hidden;
+    }
+  }
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    box-sizing: border-box;
+    height: $square-size;
+    width: $square-size;
+    border: 1px solid black;
+    background-color: #eee;
```

Comment:
It is a good practice to make all colors [variable](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties)